src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -12,166 +12,233 @@ import io.iohk.ethereum.network.EtcPeerManagerActor.PeerInfo
+  implicit val scheduler: Scheduler)
   extends Actor with ActorLogging with PeerListSupport with BlacklistSupport {
 
   import RegularSync._
   import syncConfig._
 
-
-  private var headersQueue: Seq[BlockHeader] = Nil
-  private var waitingForActor: Option[ActorRef] = None
-  var resolvingBranches: Boolean = false
-  /** This is used as an optimisation to avoid handling broadcast messages when we haven't reached the top of the chain.
-    * Currently it's set to true after we receive 0 headers from a peer, which usually means it doesn't have any new headers.
-    * But there could be other reasons for receiving 0 blocks. It may be better to make handling broadcast messages
-    * dependent on our current best block info (stored in this actor to avoid DB lookups).
-    */
-  var topOfTheChain: Boolean = false
-  private var resumeRegularSyncTimeout: Option[Cancellable] = None
-
-  var missingStateNodeRetry: Option[MissingStateNodeRetry] = None
-
   scheduler.schedule(printStatusInterval, printStatusInterval, self, PrintStatus)
 
   peerEventBus ! Subscribe(MessageClassifier(Set(NewBlock.code, NewBlockHashes.code), PeerSelector.AllPeers))
 
-  def handleCommonMessages: Receive = handlePeerListMessages orElse handleBlacklistMessages
-
   override def receive: Receive = idle
 
   def idle: Receive = handleCommonMessages orElse {
     case Start =>
       log.info("Starting block synchronization")
       appStateStorage.fastSyncDone()
-      context become running
-      askForHeaders()
+      context become running(None, Seq.empty, topOfTheChain = false, resolvingBranches = false, None, None)
+      askForHeaders(None, Seq.empty, topOfTheChain = false, resolvingBranches = false, None, None)
 
     case StartIdle =>
       appStateStorage.fastSyncDone()
-      context become running
+      context become running(None, Seq.empty, topOfTheChain = false, resolvingBranches = false, None, None)
   }
 
-  def running: Receive = handleBasicMessages orElse handleAdditionalMessages orElse handleResumingAndPrinting
-
-  def handleBasicMessages: Receive = handleCommonMessages orElse handleResponseToRequest
+  def handleCommonMessages: Receive = handlePeerListMessages orElse handleBlacklistMessages
 
-  def handleResumingAndPrinting: Receive = {
+  /**
+    * @param waitingForActor when on top of the chain and handling newBlockHashes message
+    * @param topOfTheChain   is used as an optimisation to avoid handling broadcast messages when we haven't reached the top of the chain.
+    *                        Currently it's set to true after we receive 0 headers from a peer, which usually means it doesn't have any new headers.
+    *                        But there could be other reasons for receiving 0 blocks. It may be better to make handling broadcast messages
+    *                        dependent on our current best block info (stored in this actor to avoid DB lookups).
+    */
+  def running(
+    waitingForActor: Option[ActorRef],
```

Comment:
In many methods you are using the same list of parameters. Maybe it will be more readable if you extract this fields to some State case class